### PR TITLE
MMT-3404: Adds serverless setup, deployment script

### PR DESCRIPTION
**Pull request by macrouch**

### CODE REVIEW

File: bin/deploy-bamboo.sh [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+EOF
+
+cat <<EOF > Dockerfile
+FROM node:18-bullseye
+COPY . /build
+WORKDIR /build
+RUN npm ci --omit=dev && npm run build
+EOF
+
+dockerTag=edsc-$bamboo_STAGE_NAME
```
Comment (macrouch) [inline bin/deploy-bamboo.sh:43]: shouldn't say edsc

File: package.json [hunk 2]
```diff
@@ -16,16 +16,25 @@
...
     "graphql": "^16.8.1",
     "postcss-scss": "^4.0.9",
     "react": "^18.2.0",
-    "react-dom": "^18.2.0"
+    "react-dom": "^18.2.0",
+    "serverless": "^3.35.2",
+    "serverless-esbuild": "^1.48.5",
+    "serverless-finch": "^4.0.3",
+    "serverless-offline": "^13.2.0",
+    "serverless-plugin-log-subscription": "^2.2.0",
```
Comment (macrouch) [inline package.json:36]: Remove

File: serverless-configs/aws-cors-configuration.yml [hunk 3]
```diff
@@ -0,0 +1,8 @@
+origin: '*'
+headers:
+  - Access-Control-Request-Headers
+  - Access-Control-Request-Methods
+  - Authorization
+  - Earthdata-ENV
```
Comment (macrouch) [inline serverless-configs/aws-cors-configuration.yml:6]: Not necessary

**Review by trevorlang — APPROVED** [review]
:shipit:

**Review by tle1989 — APPROVED** [review]

**Review by dmistry1 — APPROVED** [review]

**Pull request merged**